ext/pdo/pdo_dbh.c

```diff
@@ -498,42 +497,47 @@ PHP_METHOD(PDO, prepare)
 		Z_PARAM_ARRAY(options)
 	ZEND_PARSE_PARAMETERS_END();
 
-	PDO_DBH_CLEAR_ERR();
 	PDO_CONSTRUCT_CHECK;
 
-	if (ZEND_NUM_ARGS() > 1 && (opt = zend_hash_index_find(Z_ARRVAL_P(options), PDO_ATTR_STATEMENT_CLASS)) != NULL) {
-		if (Z_TYPE_P(opt) != IS_ARRAY || (item = zend_hash_index_find(Z_ARRVAL_P(opt), 0)) == NULL
-			|| Z_TYPE_P(item) != IS_STRING
-			|| (pce = zend_lookup_class(Z_STR_P(item))) == NULL
-		) {
-			pdo_raise_impl_error(dbh, NULL, "HY000",
-				"PDO::ATTR_STATEMENT_CLASS requires format array(classname, array(ctor_args)); "
-				"the classname must be a string specifying an existing class"
-				);
-			PDO_HANDLE_DBH_ERR();
-			RETURN_FALSE;
+	if (statement_len == 0) {
+		zend_argument_value_error(1, "cannot be empty");
```

Comment:
Hm, where was this enforced previously?

Reply:
No, but a test had a comment about how weird it is that you can prepare an empty SQL query.

Reply:
Okay, let's go with it and see if someone complains. It seems that at least `PDO::exec()` *did* enforce this already, so it makes sense that PDO::prepare(), PDO::query() and PDO::exec() should exhibit the same behavior.